pkg/controller/listers.go

```diff
@@ -205,14 +217,29 @@ func (l *listers) RunAsync(stopCh <-chan struct{}) {
 	go l.secretInformer.Run(stopCh)
 	go l.configMapInformer.Run(stopCh)
 	go l.podInformer.Run(stopCh)
-	synced := cache.WaitForCacheSync(stopCh,
-		l.ingressInformer.HasSynced,
-		l.endpointInformer.HasSynced,
-		l.serviceInformer.HasSynced,
-		l.secretInformer.HasSynced,
-		l.configMapInformer.HasSynced,
-		l.podInformer.HasSynced,
-	)
+	var synced bool
+	if l.enableEndpointSlicesAPI {
+		go l.endpointSliceInformer.Run(stopCh)
+		synced = cache.WaitForCacheSync(stopCh,
+			l.ingressInformer.HasSynced,
+			l.endpointSliceInformer.HasSynced,
+			l.serviceInformer.HasSynced,
+			l.secretInformer.HasSynced,
+			l.configMapInformer.HasSynced,
+			l.podInformer.HasSynced,
+		)
+
+	} else {
+		go l.endpointInformer.Run(stopCh)
```

Comment:
Remove the endpointInformer.Run() outside this if/else, which should panic if endpointslices is enabled.